pkg/pillar/hypervisor/hypervisor.go

```diff
@@ -93,9 +94,18 @@ func BootTimeHypervisor() Hypervisor {
 // the advice of this function and always ask for the enabled one.
 func GetAvailableHypervisors() (all []string, enabled []string) {
 	all = hypervisorPriority
```

Comment:
Based on a [search of the eve codebase](https://github.com/search?q=repo%3Alf-edge%2Feve%20GetAvailableHypervisors&type=code), as far as I can tell, we use this function (other than in a single test), exactly twice. Both times, we only really make use of the enabled hypervisor.

Could we just replace this messy logic with, `GetCurrentHypervisor()`?

Reply:
But ctx.hypervisorPtr is generated from allhypervisor. And that is used in couple of other places. If we want to clean up that should be a separate PR IMO.